### PR TITLE
Update default propagators and allow overrides

### PR DESCRIPTION
**Pull request by owais**
Changes:

- Default propagator is now a composite propagator comprised of B3 and
tracecontext propagators.
- Options now accepts a `propagatorFactory` function that can be used to
provide a custom propagator.
- Replaced `provider.register()` which is not part of Otel API with
explicit calls to register provider and context manager.

### CODE REVIEW

File: src/options.ts [hunk 1]
```diff
@@ -46,6 +51,7 @@ export interface Options {
   tracerConfig: NodeTracerConfig;
   spanExporterFactory: SpanExporterFactory;
   spanProcessorFactory: SpanProcessorFactory;
+  propagatorFactory: PropagatorFactory;
```
Comment (owais) [inline src/options.ts:54]: I don't imagine propagators will need to be initialized lazily or will have any dependency on config or other components but for the sake of consistency I followed factory pattern.

File: package.json [hunk 2]
```diff
@@ -79,6 +79,7 @@
     "@opentelemetry/propagator-b3": "^0.18.0",
     "@opentelemetry/resources": "^0.18.0",
     "@opentelemetry/tracing": "^0.18.0",
-    "jaeger-client": "^3.15.0"
+    "jaeger-client": "^3.15.0",
+    "semver": "^7.1.3"
```
Comment (owais) [inline package.json:83]: This is already an indirect dependency of @opentelemetry/node

File: src/options.ts [hunk 3]
```diff
@@ -115,3 +124,13 @@ export function defaultSpanExporterFactory(options: Options): JaegerExporter {
 export function defaultSpanProcessorFactory(options: Options): SpanProcessor {
   return new BatchSpanProcessor(options.spanExporterFactory(options));
 }
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function defaultPropagatorFactory(options: Options): TextMapPropagator {
+  return new CompositePropagator({
+    propagators: [
+      new B3Propagator({ injectEncoding: B3InjectEncoding.MULTI_HEADER }),
```
Comment (seemk) [inline src/options.ts:132]: Just to educate myself: is b3 multi the default propagator for Splunk distros? And why are both used?

File: src/tracing.ts [hunk 4]
```diff
@@ -31,19 +35,25 @@ export function startTracing(opts: Partial<Options> = {}): void {
 
   _patchJaeger(options.maxAttrLength);
 
-  propagation.setGlobalPropagator(
-    new B3Propagator({ injectEncoding: B3InjectEncoding.MULTI_HEADER })
-  );
+  // propagator
```
Comment (seemk) [inline src/tracing.ts:38]: nit: do we need these comments?

**Review by seemk — APPROVED** [review]

File: src/options.ts (hunk 3 above)
Comment (iNikem) [reply src/options.ts:132]: >is b3 multi the default propagator for Splunk distros

Yes

**Review by johnbley — APPROVED** [review]

File: src/options.ts (hunk 3 above)
Comment (owais) [reply src/options.ts:132]: > And why are both used?

In order to be compatible with both SFX libs and any other non-Splunk (or differently configured) Otel libs, and provide an easier migration path I'd say.

**Pull request merged**